frontend/pages/admin/IntegrationsPage/cards/CertificateAuthorities/components/DeleteCertificateAuthorityModal/helpers.ts

```diff
@@ -0,0 +1,125 @@
+  ICertificatesIntegrationDigicert,
+  IGlobalIntegrations,
+} from "interfaces/integration";
+import { useContext } from "react";
+import { IDigicertFormData } from "../DigicertForm/DigicertForm";
+
+export const useCertAuthorityDataGenerator = (
+  certAuthorityType: ICertificateAuthorityType,
+  certAuthority?: ICertificateIntegration
+) => {
+  const { config } = useContext(AppContext);
+
+  const generateAddPatchData = (formData: IDigicertFormData) => {
+    if (!config) return null;
+
+    const data: { integrations: Partial<IGlobalIntegrations> } = {
+      integrations: {},
+    };
+
+    switch (certAuthorityType) {
+      case "ndes":
+        break;
+      case "digicert":
+        data.integrations.digicert = [
+          ...(config.integrations.digicert || []),
+          {
+            name: formData.name,
+            url: formData.url,
+            api_token: formData.apiToken,
+            profile_id: formData.profileId,
+            certificate_common_name: formData.commonName,
+            certificate_user_principal_names: [formData.userPrincipalName],
+            certificate_seat_id: formData.certificateSeatId,
+          },
+        ];
+        break;
+      case "custom":
+        break;
+      default:
+        break;
+    }
+
+    return data;
+  };
+
+  /**
+   * generates the data to be sent to the API to delete the certificate authority.
+   * under the hood we are updating the app config object with the new data and
+   * have to generate the correct data for the PATCH request.
+   */
+  const generateDeletePatchData = () => {
+    if (!config) return null;
+
+    const data: { integrations: Partial<IGlobalIntegrations> } = {
+      integrations: {},
+    };
+
+    switch (certAuthorityType) {
+      case "ndes":
+        data.integrations.ndes_scep_proxy = null;
+        break;
+      case "digicert":
+        data.integrations.digicert = config.integrations.digicert?.filter(
+          (cert) => {
+            return (
+              (certAuthority as ICertificatesIntegrationDigicert).name !==
+              cert.name
+            );
+          }
+        );
+        break;
+      case "custom":
+        data.integrations.custom_scep_proxy = config.integrations.custom_scep_proxy?.filter(
+          (cert) => {
+            return (
+              (certAuthority as ICertificatesIntegrationCustomSCEP).id ===
+              cert.id
+            );
+          }
+        );
+        break;
+      default:
+        break;
+    }
+
+    return data;
+  };
+
+  /**
+   * generates the data to be sent to the API to edit the certificate authority.
+   * under the hood we are updating the app config object with the new data and
+   * have to generate the correct data for the PATCH request.
+   */
+  const generateEditPatchData = (formData: IDigicertFormData) => {
+    if (!config) return null;
+  };
+
+  return {
+    generateAddPatchData,
+    generateDeletePatchData,
+    generateEditPatchData,
```

Comment:
I think these function references might get recreated as the custom hook fires on each render. Not sure if that's a concern in the implementation, but if so, I believe you could wrap the functions in useCallBack before returning.

Reply:
will do that now.